### PR TITLE
gradle 4.0.2

### PR DESCRIPTION
**Pull request by seanjreilly**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/gradle.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Gradle < Formula
   desc "Build system based on the Groovy language"
   homepage "https://www.gradle.org/"
-  url "https://services.gradle.org/distributions/gradle-4.0.1-all.zip"
-  sha256 "8a8005633be0ca38a206f2590445685683452a0b98a5d8ffd5b8413be09bf998"
+  url "https://services.gradle.org/distributions/gradle-4.0.2-bin.zip"
```
Comment (fxcoudert) [inline Formula/gradle.rb]: Needs the `all` package, not only binaries

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by seanjreilly** [comment]
If we require the all package, not just the binaries, then shouldn't there be a test that fails when somebody bumps the formula without using -all? All of the tests passed when I used `brew bump-formula-pr`, so it wasn't obvious that I was doing the wrong thing.

File: Formula/gradle.rb (hunk 1 above)
Comment (seanjreilly) [inline Formula/gradle.rb]: Unfortunately, this change isn't sufficient. The sha256 hash is different between the two versions of the binary.

**Comment by seanjreilly** [comment]
@fxcoudert I've rebased with the url change you requested, plus the SHA change that is needed for it to work. Can this be merged now?

Also, if we require the all package, not just the binaries, then shouldn't there be a test that fails when somebody bumps the formula without using -all? All of the tests passed when I used brew bump-formula-pr, so it wasn't obvious that I was doing the wrong thing.

**Comment by fxcoudert** [comment]
I think the all package is required when using `--with-all`, which is not on by default and thus not tested.

Thanks for updating the PR!

**Review by fxcoudert — APPROVED** [review]

**Comment by seanjreilly** [comment]
If it's required for something, shouldn't it be tested? This is likely to keep happening otherwise.

File: Formula/gradle.rb [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
   desc "Build system based on the Groovy language"
   homepage "https://www.gradle.org/"
-  url "https://services.gradle.org/distributions/gradle-4.0.1-all.zip"
-  sha256 "8a8005633be0ca38a206f2590445685683452a0b98a5d8ffd5b8413be09bf998"
-
-  devel do
-    url "https://services.gradle.org/distributions/gradle-4.1-rc-1-all.zip"
-    version "4.1-rc-1"
-    sha256 "eb504c25abae8ee43e01a56f6873137d74ce68c43b8b1fe89dc0562e6b81ca51"
-  end
```
Comment (indrajitr) [inline Formula/gradle.rb]: devel build should be retained. Raised #16168 for that.

**Pull request merged**